app/app.go

```diff
@@ -57,17 +57,24 @@ func NewApp() (*App, error) {
 
 	wrappedPolicy := newWrappedPolicy(existingPolicy)
 
-	paused := false // always starts off running
+	var isRunning bool
+	if b, err := store.GetLatestRunningState(); err != nil {
+		isRunning = true
+		logging.Error(err.Error())
+	} else {
+		isRunning = b
+	}
+
 	return &App{
 		vc:     vaultClient,
 		wp:     wrappedPolicy,
-		paused: &paused,
+		isRunning: &isRunning,
 		router: NewRouter(
 			&endpoints{
 				wp:     wrappedPolicy,
 				vc:     vaultClient,
 				store:  store,
-				paused: &paused,
+				isRunning: &isRunning,
```

Comment:
A bit lor sor but to add on to the previous argument; here if you accidentally mutate the value of the external `isRunning`, it actually affects the value "stored" in `App`, because what App does is merely to hold the reference of the external `isRunning` created here, which is probably not what you really want to do.

Reply:
i do want to mutate the value stored in App since it controls the actual scaling

Reply:
Okay noted the possible concurrent mutation from `App` and `Router` side